torch/_guards.py

```diff
@@ -605,10 +606,13 @@ def restore_graphstate(self, state: GlobalContextCheckpointState) -> None:
 # Like a Set[Guard] but will record the user stack on all guards at the
 # time they were installed at their destination
 class GuardsSet:
-    def __init__(self, inner: Optional[set[Guard]] = None) -> None:
+    def __init__(self, inner: Optional[OrderedSet[Guard] | set[Guard]] = None) -> None:
```

Comment:
Can you check where we are calling with `set` and then use `OrderedSet` there as well? We can put an assertion here that inner is either None or OrderedSet.